assets/js/edd-checkout-global.js

```diff
@@ -146,7 +146,16 @@ window.EDD_Checkout = (function($) {
 
 						var inputs = $('#edd_cc_fields .edd-input, #edd_cc_fields .edd-select,#edd_cc_address .edd-input, #edd_cc_address .edd-select,#edd_payment_mode_select .edd-input, #edd_payment_mode_select .edd-select');
 
-						if( '0.00' == discount_response.total_plain ) {
+						if( true == discount_response.require_payment_info ) {
+
+							if (!inputs.is('.card-address-2')) {
```

Comment:
@tnorthcutt can you explain why this is necessary?

Reply:
Sure! That's what keeps the credit card form from being hidden (and not required), IF a filter has been added to require payment info even on 100% discounted carts. That filter can be added like so:

```
add_filter( 'edd_require_payment_info', '__return_true' );
```

Without adding that filter, behavior is completely "normal".